scripts/compose.py

```diff
@@ -1168,6 +1168,46 @@ def _content(self):
         return content
 
 
+class OpbeansLoadGenerator(Service):
+    @classmethod
+    def add_arguments(cls, parser):
+        super(OpbeansLoadGenerator, cls).add_arguments(parser)
+        for service_class in OpbeansService.__subclasses__():
+            parser.add_argument(
+                '--no-%s-loadgen' % service_class.name(),
+                action='store_true',
+                default=False,
+                help='Disable load generator for {}'.format(service_class.name())
+            )
+
+    def __init__(self, **options):
+        super(OpbeansLoadGenerator, self).__init__(**options)
+        self.loadgen_services = []
+        # create load for opbeans services (identified via the disable_opbeans_XYZ flag)
+        # unless the no_opbeans_XYZ_flag is set
+        for flag, value in options.items():
+            if not value and flag.startswith('disable_opbeans_'):
+                service_name = flag[len('disable_'):]
+                if not options.get('no_{}_loadgen'.format(service_name)) and service_name != 'opbeans_load_generator':
+                    self.loadgen_services.append(service_name.replace('_', '-'))
+
+    @staticmethod
+    def enabled():
+        return True
```

Comment:
I don't think this should be a default service - we want it to be enabled by default if any opbeans service is - that could be done he, like with postgres and redis: re https://github.com/elastic/apm-integration-testing/blob/cdd26cd2d4cb59eef5f80409bde0b45057d18219/scripts/compose.py#L2128-L2132

Reply:
to be honest, I'm not a huge fan how we (I think it was me in the original `local` script) solved this issue here. Depending on the existence of some flags, we add/remove some other flags at run-time. That seems wrong.

Not sure what a better solution would be. For the `--no` / `--with` flags, maybe we can simply always add them. Postgres/Redis and now load gen, I'm not so sure. The load gen container will immediately exit if there's no opbeans services running, so I don't think it would be terribly bad if it always started

Reply:
I agree it's not the best solution but I think `argparse` puts us in a tough spot here (or I missed something).
The problem is that integration tests will start this and wait for all services to be healthy, and this won't be so they abort.  We either need to change this or update all of the wrapper scripts not to start it.